OEPS/OEP-5.mediawiki

```diff
@@ -103,25 +106,55 @@ If any of the transfers fail, all of the transfers must be failed, and the metho
 def approve(to, tokenId)
 </pre>
 
-
-The approve function allows the <code>to</code> address to transfer the <code>tokenId</code> NFT.
+The approve method allows the <code>to</code> address to transfer the <code>tokenId</code> NFT.
 If this function is called again it overwrites the <code>to</code> address that can transfer the NFT.
 <code>tokenId</code> should be string or bytearray.
 The <code>to</code> address must be 20-byte address.
 If not, this method must <code>throw</code> an exception.
 
-====takeOwnership====
+====allowance====
+
+<pre>
+def allowance(tokenId)
+</pre>
+
+Returns the address that is allowed to send the NFT with id <code>tokenId</code>.
+
+===Optional Methods===
+
+====clearApproved====
 
 <pre>
-def takeOwnership(to, tokenId)
+def clearApproved(tokenId)
 </pre>
 
-Either the token owner or the approved account can assign the ownership of the NFT with the id <code>tokenId</code> to the <code>to</code> address.
+The clearApproved method removes the address that was set by the approve method for the NFT with the id <code>tokenId</code>.
 <code>tokenId</code> should be string or bytearray.
-<code>to</code> must be 20-byte addresses.
+This method can only be called by the token owner.
 If not, this method must <code>throw</code> an exception.
 
-===Optional Methods===
+====approvalForAll====
+
+<pre>
+def approvalForAll(owner, to, approval)
```

Comment:
This interface seems be duplicated with approve(to, tokenId)?

By approving to current owner of TokenID, the previous approval can also be withdrawn.
Any reason for this interface?

Reply:
`approvalForAll` is similar, but not the same. It allows someone to approve another address to manage all of its tokens. The `approve` method only approves control over 1 specific token.

I added this based on @tanZiWen's suggestion - it's part of ERC721: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md

Reply:
ok